packages/shared_preferences/shared_preferences/tool/pre_publish.dart

```diff
@@ -0,0 +1,39 @@
+import 'dart:io';
+
+Future<void> _runCommand({
+  required String message,
+  required String executable,
+  required List<String> arguments,
+}) async {
+  stdout.write(message);
+  final ProcessResult pubGetResult = await Process.run(
+    executable,
+    arguments,
+    workingDirectory: '../shared_preferences_tool',
```

Comment:
@stuartmorgan I'm not sure if this is the right relative path. I didn't get if the script runs in the `shared_preferences` directory or in the `shared_preferences/tool` director.

Reply:
Never mind, it actually worked:

<img width="1835" alt="image" src="https://github.com/user-attachments/assets/bef01200-b455-44dc-b182-17f11193895d">

Reply:
Please don't rely on relative paths; see [here](https://github.com/flutter/packages/blob/f142461af2fa39f8121ddfbbd0720dfafb016996/packages/web_benchmarks/tool/run_tests.dart#L25-L26) for a method that is not sensitive to the caller's working directory.

Reply:
@stuartmorgan I missed this comment.

Just pushed the suggested changes.